lib/mongoid/threaded.rb

```diff
@@ -197,13 +197,17 @@ def client_override=(name)
     # Get the current Mongoid scope.
     #
     # @example Get the scope.
+    #   Threaded.current_scope(klass)
     #   Threaded.current_scope
     #
+    # @param [ Klass ] klass The class type of the scope.
+    #
     # @return [ Criteria ] The scope.
     #
     # @since 5.0.0
-    def current_scope
-      Thread.current[CURRENT_SCOPE_KEY]
+    def current_scope(klass = nil)
+      return Thread.current[CURRENT_SCOPE_KEY] unless klass && Thread.current[CURRENT_SCOPE_KEY] && Thread.current[CURRENT_SCOPE_KEY].is_a?(Hash)
+      Thread.current[CURRENT_SCOPE_KEY][ Thread.current[CURRENT_SCOPE_KEY].keys.find { |k| k <= klass } ]
     end
```

Comment:
We can split this into two conditions:
1) is klass not nil and is Thread.current[CURRENT_SCOPE_KEY] a hash? -> return line 210
2) return Thread.current[CURRENT_SCOPE_KEY]

So another way of writing this would be

```ruby
    def current_scope(klass = nil)
      if klass && Thread.current[CURRENT_SCOPE_KEY].respond_to?(:keys)
        Thread.current[CURRENT_SCOPE_KEY][
            Thread.current[CURRENT_SCOPE_KEY].keys.find { |k| k <= klass }
        ]
      else
        Thread.current[CURRENT_SCOPE_KEY]
      end
    end
```